### DISCUSSION TITLE
Can we hook into the workspace save workflow when the user tries to save the workspace?

### DISCUSSION THREAD

**Discussion by JoshwinThomasIBM** [microsoft/vscode-discussions#2457] [Extension Development QnA] [answered]
When closing a workspace (via File -> Close Workspace), a prompt asks the user whether they want to save the workspace. Can we hook into the flow when the user selects an option here? Additionally, if the user opts to save, a new dialog box appears for the workspace name and location. Can we also hook into this flow based on the user’s selection?

Similar to the above case, when the user tries to close the entire IDE, they are asked whether they want to save the workspace as above.  Can we intercept this prompt and manage the flow?

**Comment by alefragnani** [comment, marked as answer]
Hi @JoshwinThomasIBM ,

There is no APIs available right now, neither for the save nor close events. So, if you really need such events, I would suggest you to open a feature request in the VS Code repo (https://github.com/microsoft/vscode/issues).

But, I wonder the use case. Why do you need a hook while saving and naming the workspace file, closing closing the IDE. Do you need to know the workspace file name when the user saves/names it; or freeing resources when the user closes VS Code.

Hope this helps
